### DISCUSSION TITLE
dlt_user.so send log with timestamp overflow

### DISCUSSION THREAD

**Discussion by jack-liuhang** [COVESA/dlt-daemon#563] [Q&A] [answered]
```c
uint32_t dlt_uptime(void)
{

#if defined (__WIN32__) || defined(_MSC_VER)

    return (uint32_t)(GetTickCount() * 10); /* GetTickCount() return DWORD */

#else
    struct timespec ts;

    if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
        return (uint32_t)ts.tv_sec * 10000 + (uint32_t)ts.tv_nsec / 100000; /* in 0.1 ms = 100 us */
    else
        return 0;

#endif
```
uint32_t maximum=4294967295
```python
val=4294967295//10000
print(val)
print(429496/3600/24) #dlt log limit in about 5 days
```
**Question:**
If we want to keep a precision with 0.1 ms, our log timestamp limit in 5days because ot  limit scope of `uint32_t`

maybe we should conside about change the precision  of log timestamp

**Comment by minminlittleshrimp** [comment]
I would like to continue this topic in discussion tag.

**Comment by minminlittleshrimp** [comment, marked as answer]
Hello @jack-liuhang 
For me, this would be the current implementation based on [R19-11](https://www.autosar.org/fileadmin/standards/R19-11/FO/AUTOSAR_PRS_LogAndTraceProtocol.pdf)
[PRS_Dlt_00323] ⌈The Timestamp field (TMSP) shall be a 32-bit unsigned integer.⌋
(RS_LT_00017)
The time resolution 0.1 ms defined at [Dlt309]
Regards

**Reply by jack-liuhang** [reply]
Thanks 
It's a pity, we cannot do anything to improve this overflow issue because of this standard
This doc is useful, thank you again.
Regards

**Comment by minminlittleshrimp** [comment]
Hello @jack-liuhang 
if you still have interest in this topic, the next Protocol v2 dlt in Autosar standard introducing timestamp that could adapt to avoid Y2038 issue. You can check it out at: [R21-11](https://www.autosar.org/fileadmin/standards/R21-11/FO/AUTOSAR_PRS_LogAndTraceProtocol.pdf)

[PRS_Dlt_01004] ⌈If the Log and Trace message is a Data Message (Verbose Mode 
or Non-Verbose Mode), the TMSP2 (Timestamp) with a nanosecond resolution shall 
be added to the Base Header.⌋ (RS_LT_00002, RS_LT_00013, RS_LT_00017)

However, DLT team currently do not plan any release for dlt major version 3.0, so I have no idea of dlt-daemon future.
Regards

**Reply by jack-liuhang** [reply]
Hi

Thank you for your reminder and notification regarding the upcoming changes. The introduction of Protocol v2 within the Autosar standard, especially with its significant alterations and the adoption of a 9-byte timestamp, indeed marks a substantial evolution. I am confident that these updates will effectively address the current issues.

Please ensure to notify me upon the release of this version. Your timely updates are greatly appreciated.

Regards
